packages/db-base/src/lib/redisHandler.ts

```diff
@@ -1,18 +1,76 @@
-const Resp = require('respjs');
-const { EventEmitter } = require('events');
-const { QUEUED_STR_BUF, OK_STR_BUF } = require('./constants');
+import type { Socket } from 'net';
+// @ts-expect-error no types
+import Resp from 'respjs';
+import { EventEmitter } from 'events';
+import { QUEUED_STR_BUF, OK_STR_BUF } from './constants';
+
+interface RedisHandlerOptions {
+    // Logger object
+    log: any;
+    // log prefix
+    logScope?: string;
+    // if data should be handled as buffer
+    handleAsBuffers: boolean;
+    // additonal debug information
+    enhancedLogging?: boolean;
+}
+
+interface WriteQueueElement {
+    // response id
+    id: ResponseId;
+    // response data, temporary false if not ready to sent yet
+    data: false | Buffer;
+}
+
+interface FullWriteQueueElement extends WriteQueueElement {
+    // response data
+    data: Buffer;
+}
+
+interface MultiCallElement {
+    // all response ids of the multi call
+    responseIds: ResponseId[];
+    // indicator if exec already called
+    execCalled: boolean;
+    // number of responses which are already ready
+    responseCount: 0;
```

Comment:
I'm not sure I understand the purpose of defining this MUST be 0 if we're incrementing it later.
After the incrementation, the type definition is lying.